cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -142,8 +143,15 @@ implement its ``run()`` method::
     }
   }
 
-The custom action then can be added to the workflow using the ``addAction()`` method as
-shown above.
+The custom action then can be added to a workflow using the ``addAction()`` method, as
+shown in the previous example::
+
+  . . .
+  addAction(new MyAction());
+  . . .
+  
+This is appropriate for long-running custom actions, as you now have control over the use
+of transactions in any database access.
```

Comment:
May be something along the line `Since you now have better control over transactions (can be started only for dataset accesses), long-running tasks can be executed in the custom actions which do not require transactions.`